File: solutions/javascript/ReplaceWords/ReplaceWords.js
/**
 * @param {string[]} dict
 * @param {string} sentence
 * @return {string}
 */
var replaceWords = function(dict, sentence) {
    var arr = sentence.split(' ');
    var res = [];
    for (var i = 0; i < arr.length; i++) {
        res.push(arr[i]);
        for (var j = 0; j < dict.length; j++) {
            if (arr[i].indexOf(dict[j]) == 0) {
                res.pop();
                res.push(dict[j]);
                break;
            }
        }
    }
    return res.join(' ');
};

File: solutions/java/ReplaceWords/ReplaceWords.java
class Solution {
    public String replaceWords(List<String> dict, String sentence) {
        String[] strings = sentence.split(" ");
        for (int i = 0; i < strings.length; i++) {
            for (String s : dict) {
                if (strings[i].indexOf(s) == 0) {
                    strings[i] = s;
                    break;
                }
            }
        }
        return String.join(" ", strings);
    }
}